server/providers/FantLab.js

```diff
@@ -0,0 +1,170 @@
+  // 57 - newspaper
+  // types can get here https://api.fantlab.ru/config.json
+  _filterWorkType = [7, 11, 12, 22, 23, 24, 25, 26, 46, 47, 49, 51, 52, 55, 56, 57];
+  _baseUrl = 'https://api.fantlab.ru'
+
+  constructor() { }
+
+  async search(title, author) {
+    var searchString = encodeURIComponent(title)
+    if (author) {
+      searchString += encodeURIComponent(' ' + author)
+    }
+    var url = `${this._baseUrl}/search-works?q=${searchString}&page=1&onlymatches=1`
+    Logger.debug(`[FantLab] Search url: ${url}`)
+    var items = await axios.get(url).then((res) => {
+      return res.data || []
+    }).catch(error => {
+      Logger.error('[FantLab] search error', error)
+      return []
+    })
+
+    return Promise.all(items.map(async item => await this.getWork(item)))
+  }
+
+  async getWork(item) {
+    var { work_id, work_type_id } = item
+
+    if (this._filterWorkType.includes(work_type_id)) {
+      return { title: null }
+    }
+
+    var url = `${this._baseUrl}/work/${work_id}/extended`
+    var bookData = await axios.get(url).then((resp) => {
+      return resp.data || null
+    }).catch((error) => {
+      Logger.error(`[FantLab] work info reques error`, error)
+      return null
+    })
+
+    return await this.cleanBookData(bookData)
+  }
+
+  async cleanBookData(bookData) {
+    var { authors, work_name_alts, work_id, work_name, work_year, work_description, image, classificatory, editions_blocks } = bookData;
+
+    var subtitle = Array.isArray(work_name_alts) ? work_name_alts[0] : null
+    var auth = authors.map(function (author) {
+      return author.name
+    });
+
+    var genres = classificatory ? this.tryGetGenres(classificatory) : []
+
+    var imageAndIsbn = await this.tryGetCoverFromEditions(editions_blocks)
+
+    if (imageAndIsbn) {
+      var { imageUrl, isbn } = imageAndIsbn
+
+      if (imageUrl) {
+        image = imageUrl
+      }
+    }
+
+    var cover = 'https://fantlab.ru' + image
+
+    return {
+      id: work_id,
+      title: work_name,
+      subtitle: subtitle || null,
+      author: auth ? auth.join(', ') : null,
+      publisher: null,
+      publishedYear: work_year,
+      description: work_description,
+      cover: image ? cover : null,
+      genres: genres,
+      isbn: isbn
+    }
+  }
+
+  tryGetGenres(classificatory) {
+    var { genre_group } = classificatory;
+    if (!genre_group) {
+      return []
+    }
+    var genresGroup = genre_group.find(group => group.genre_group_id = 1) // genres and subgenres
+
+    // genre_group_id=2 - General Characteristics
+    // genre_group_id=3 - Arena
+    // genre_group_id=4 - Duration of action
+    // genre_group_id=6 - Story moves
+    // genre_group_id=7 - Story linearity
+    // genre_group_id=5 - Recommended age of the reader
+
+    if (!genresGroup) return []
+
+    var { genre } = genresGroup;
+    var rootGenre = genre[0];
+
+    var { label } = rootGenre
+
+    return [label].concat(this.tryGetSubGenres(rootGenre))
```

Comment:
Do we need to separate this out into a sub genres function?
Could we get them all like this `genre.map(genreObj => genreObj.label)`

Reply:
![image](https://user-images.githubusercontent.com/13904687/217634479-6cdebbd9-ebe5-4f66-a7ab-6cd7ca19d33b.png)
root genre and sub genres has same node name